sync_auditlog/models/log.py

```diff
@@ -0,0 +1,237 @@
+# Copyright (C) 2021 Open Source Integrators
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+from datetime import datetime
+
+try:
+    from xmlrpc import client as xmlrpclib
+except ImportError:
+    import xmlrpclib
+
+
+class AuditlogLog(models.Model):
+    _inherit = "auditlog.log"
+
+    state = fields.Selection(
+        [
+            ("logged", "Logged"),
+            ("captured", "Captured"),
+            ("Prepared", "Prepared"),
+            ("Pushed", "Pushed"),
+            ("Pulled", "Pulled"),
+            ("Processed", "Processed"),
+            ("Error", "Error"),
+            ("Failed", "Failed"),
+            ("Cancelled", "Cancelled"),
+        ],
+        string="State",
+        default="logged",
+    )
+    uuid = fields.Char("UUID")
+    timestamp = fields.Datetime("Timestamp")
+    resource_ids = fields.Char("Resource IDs")
+    raw_args_kwargs = fields.Char("Raw Args and kwargs")
+    prepared_args_kwargs = fields.Char("Prepared Args and kwargs")
+    context = fields.Char("Context")
+    external_id = fields.Char("External ID")
+    parent_log = fields.Char("Parent log")
+    result = fields.Char("Result")
+    remote = fields.Char("Remote")
+    message = fields.Char("Message")
+    _sql_constraints = [
+        (
+            "uuid_uniq",
+            "unique(uuid)",
+            (
+                "This UUID already exists\n"
+                "Record will be skipped to avoid duplication"
+            ),
+        )
+    ]
+
+    def _cron_prepare_auditlog_events(self):
+        logs = self.search([("state", "=", "captured")])
+        for log in logs:
+            # ToDo - Prepare data Logic
+            print("============ToDo Prepare Data Logic============")
+
+    def _push_event_data(self, addr, uid, password, dbname):
+
+        try:
+            model = xmlrpclib.ServerProxy("%s/xmlrpc/object" % (addr)).execute(
```

Comment:
I'm quite sure we can skip this call. AFAICS it is checking the log model exists in the server. This can be also checked when try-ing the execute call.

Reply:
I have added this call as we require the Model ID to create logs in the remote system. As the model database ID can be different in the remote system, I am fetching the model_id by using the name of the model and then using it in the dictionary to create the log entry.

Reply:
Got it. Then I suggest adding a `model` char field where you can write the model identifier ("res.partner").
Doing so would trigger logic populatig the model_id on that system.
That way we can avoid the Model DB ID issue and simplify the upload.